proposals/tf_utf8_identifiers/README.md

```diff
@@ -0,0 +1,435 @@
+
+The path lexer is used as the first step in validating prim path strings.  The text file format lexer is used to read and validate the content of text-based usd files (i.e., `.usda`).  Both require that a sequence of characters be recognized as a logical token (and parsed in context) before being validated.  The rules that dictate what lexer sequences are acceptable for identifiers need to be updated accordingly for Unicode based identifiers.  It is desireable that these rules conform to those updated in the code-based validation sequences.  Unfortunately, it is difficult in flex rules to specify the entire set of byte strings staht would encapsulate a character in the e.g., `XID_Start` character class.  To get around this, we propose the lex rule be very permissive (basically any properly encoded UTF-8 character) and specifc validation (prim name or identifier) on the lexed string be done as part of the lexer logic to accept the token prior to parsing.  This does have the unfortunate overhead of performing some validations twice, once at tokenization using the state machine and once adding the token to the tree and checking for identifer validity.
+
+One complication is the use of flex and bison to generate the state machines and code necessary to lex and parse these sequences.  While we would like to limit changes such that only flex is required to regenerate the code and state machine, it is necessary to also regenerate the parser generator through bison.  Unfortunately, different versions of flex and bison generate code in slightly different ways.  In particular, a few function signatures change depending on the version used.  Additionally, flex and bison are natively Linux-based tools - to support Windows, the use of winflex / winbison is required, so it is necessary to lock down a common version that is supported across both platforms.
+
+Note that all of the above relies on the source data being UTF-8 encoded.  This should not be a problem for legacy ASCII based `usda` files or code and should maintain backward compatibility with no changes required (as long as extended character sets are not used).  Moving forward, any USD content that wants to take advantage of Unicode support must encode content in UTF-8.  This includes both files (text and binary) as well as strings in source code.  While these values can be handled correctly in USD, proper display of these strings will be application dependent.  To support testing UTF-8, several existing tests were saved such that content was UTF-8 encoded.  This applies both to USD files with Unicode content and code with string embedded for specific testing of Unicode identifier features.
```

Comment:
Since we've never seen a bump, I don't know the right answer to that question.  It makes logical sense and perhaps we should include the suggestion in the proposal (especially in a diffing scenario where the ordering really does differ in ASCII properties between a version 1.0 file and a version 2.0 file).

Reply:
Yeah.  I'm thinking something like `#usda 1.0` files were strictly ASCII but `#usda 1.1` files are strictly UTF-8.

Reply:
added additional content to discuss bumping the version